firewood/src/merkle/stream.rs

```diff
@@ -1,25 +1,22 @@
 // Copyright (C) 2023, Ava Labs, Inc. All rights reserved.
 // See the file LICENSE.md for licensing terms.
 
-use super::{node::Node, LeafNode, Merkle, MerkleError, NodeType, ObjRef};
+use super::{node::Node, BranchNode, Merkle, NodeType, ObjRef};
 use crate::{
     shale::{DiskAddress, ShaleStore},
     v2::api,
 };
 use futures::Stream;
-use std::task::Poll;
+use std::{ops::Not, task::Poll};
 
 pub(super) enum IteratorState<'a> {
     /// Start iterating at the beginning of the trie,
     /// returning the lowest key/value pair first
     StartAtBeginning,
     /// Start iterating at the specified key
     StartAtKey(Vec<u8>),
-    /// Continue iterating after the given last_node and parents
-    Iterating {
-        last_node: ObjRef<'a>,
-        parents: Vec<(ObjRef<'a>, u8)>,
-    },
+    /// Continue iterating after the given `next_node` and parents
+    Iterating { parents: Vec<(ObjRef<'a>, u8)> },
```

Comment:
I can't add a comment to line 30 below, which is the implementation of Default for IteratorState. I couldn't find a use case where you're using this in the new code, so I believe the code that implements `Default` can be removed.

Reply:
removed